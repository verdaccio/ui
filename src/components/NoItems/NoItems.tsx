import React from 'react';
import Typography from '@material-ui/core/Typography';

interface Props {
  text: string;
  className?: string;
}

const NoItems: React.FC<Props> = ({ className, text }) => (
  <Typography className={className} gutterBottom={true} variant="subtitle1">
    {text}
  </Typography>
);

export default NoItems;
