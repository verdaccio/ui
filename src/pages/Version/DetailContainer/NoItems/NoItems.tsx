import React from 'react';

import Text from 'verdaccio-ui/components/Text';

interface Props {
  text: string;
  className?: string;
}

const NoItems: React.FC<Props> = ({ className, text }) => (
  <Text className={className} gutterBottom={true} variant="subtitle1">
    {text}
  </Text>
);

export default NoItems;
