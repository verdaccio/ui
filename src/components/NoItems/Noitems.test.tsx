import React from 'react';
import { shallow, mount } from 'enzyme';

import NoItems from './NoItems';

console.error = jest.fn();

describe('<NoItem /> component', () => {
  const props = {
    text: 'test',
  };

  test('should load the component in default state', () => {
    const wrapper = shallow(<NoItems text={props.text} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should set html from props', () => {
    const props = {
      text: 'This is a test string',
    };
    const wrapper = mount(<NoItems text={props.text} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
