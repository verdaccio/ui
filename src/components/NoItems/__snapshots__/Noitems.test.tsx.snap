// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoItem /> component should load the component in default state 1`] = `
<h6
  class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
>
  test
</h6>
`;
